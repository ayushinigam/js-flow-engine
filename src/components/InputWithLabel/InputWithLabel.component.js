import React from 'react';
import './InputWithLabel.styles.css';

const InputWithLabel = ({className, label, inputClassName, value, ...extraprops}) => (
  <div className={`input-with-label ${className}`}>
    <p>{label}</p>
    <input className={inputClassName} value={String(value)} {...extraprops} />
  </div>
);

export default InputWithLabel;
