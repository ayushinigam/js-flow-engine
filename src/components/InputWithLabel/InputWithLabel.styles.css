.input-with-label {
  display: block;
  padding: 10px; }
  .input-with-label input {
    border: none;
    padding: 10px;
    font-weight: bold;
    background-color: #DADBDA;
    color: black;
    width: 100%; }
  .input-with-label p {
    padding-bottom: 10px; }
