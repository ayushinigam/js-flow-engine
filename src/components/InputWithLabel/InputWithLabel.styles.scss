@import "../../variables.scss";


.input-with-label {
  display: block;
  padding: 10px;
  input {
    border: none;
    padding: 10px;
    font-weight: bold;
    background-color: #DADBDA;
    color: black;
    width: 100%;
  }
  p {
    padding-bottom: 10px;
  }
}
