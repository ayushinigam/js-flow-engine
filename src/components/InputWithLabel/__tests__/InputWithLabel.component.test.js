import React from 'react';
import renderer from 'react-test-renderer';
import InputWithLabel from '../InputWithLabel.component';
describe('InputWithLabel component', () => {
  it('InputWithLabel: renders correctly', () => {
    const tree = renderer.create(<InputWithLabel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
