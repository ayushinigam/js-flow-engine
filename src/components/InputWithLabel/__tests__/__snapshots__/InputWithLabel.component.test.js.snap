// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel component InputWithLabel: renders correctly 1`] = `
<div
  className="input-with-label undefined"
>
  <p />
  <input
    className={undefined}
    value="undefined"
  />
</div>
`;
